cmd/cloudFoundryCreateService_test.go

```diff
@@ -0,0 +1,128 @@
+package cmd
+
+import (
+	"testing"
+
+	"github.com/SAP/jenkins-library/pkg/mock"
+	"github.com/SAP/jenkins-library/pkg/telemetry"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestCloudFoundryCreateService(t *testing.T) {
+	execRunner := mock.ExecMockRunner{}
+	var telemetryData telemetry.CustomData
+	t.Run("Create service: no broker, no config, no tags", func(t *testing.T) {
+		config := cloudFoundryCreateServiceOptions{
+			CfAPIEndpoint:         "https://api.endpoint.com",
+			CfOrg:                 "testOrg",
+			CfSpace:               "testSpace",
+			Username:              "testUser",
+			Password:              "testPassword",
+			CfService:             "testService",
+			CfServiceInstanceName: "testName",
+			CfServicePlan:         "testPlan",
+		}
+		error := runCloudFoundryCreateService(&config, &telemetryData, &execRunner)
+		if error == nil {
+			assert.Equal(t, "cf", execRunner.Calls[0].Exec)
+			assert.Equal(t, []string{"create-service", "testService", "testPlan", "testName"}, execRunner.Calls[0].Params)
+		}
+	})
+	t.Run("Create service: only tags", func(t *testing.T) {
+		config := cloudFoundryCreateServiceOptions{
+			CfAPIEndpoint:         "https://api.endpoint.com",
+			CfOrg:                 "testOrg",
+			CfSpace:               "testSpace",
+			Username:              "testUser",
+			Password:              "testPassword",
+			CfService:             "testService",
+			CfServiceInstanceName: "testName",
+			CfServicePlan:         "testPlan",
+			CfServiceTags:         "testTag, testTag2",
+		}
+		error := runCloudFoundryCreateService(&config, &telemetryData, &execRunner)
+		if error == nil {
+			assert.Equal(t, "cf", execRunner.Calls[1].Exec)
```

Comment:
Are we defining here dependencies wrt order of the tests. E.g. when I add some tests inbetween I have to adjust the index for all the tests down the road? I think we should avoid this.

Reply:
Removed the dependencies wrt order of the tests and introduced `cfMockCleanup` function for that purpose.